src/components/editor/Bindings/Store/create.ts

```diff
@@ -98,12 +101,33 @@ const getInitialState = (
     ...getInitialStateData(),
 
     initializeCollectionData: (currentCollection, persistedDraftId) => {
-        const { setCollectionData } = get();
+        const { setCollectionData, setSchemaInferenceDisabled } = get();
 
         if (currentCollection) {
             evaluateCollectionData(persistedDraftId, currentCollection).then(
-                (response) => setCollectionData(response),
-                () => setCollectionData(undefined)
+                (response) => {
+                    setCollectionData(response);
+
+                    if (response) {
+                        const writeSchemaKey = response.spec.hasOwnProperty(
+                            'writeSchema'
+                        )
+                            ? 'writeSchema'
+                            : 'schema';
+
+                        const inferenceAnnotationExists = !response.spec[
+                            writeSchemaKey
+                        ].hasOwnProperty(Annotations.inferSchema);
+
+                        setSchemaInferenceDisabled(inferenceAnnotationExists);
```

Comment:
Why are we checking if the property does not exist and then store it in a variable stating it does exist? Also - what is driving not checking the actual value of the prop and only the exists of the key?

Reply:
Q1: Typo. If you look at the commit history, the property used to be named `schemaInferenceEnabled`.

Q2: That can be done. In practice, the annotation is applied in a similar fashion to an optional argument that can only have the value of `true`. Will record the value explicitly _just in case_ a change in annotation application occurs.

Reply:
Should be resolved.